python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt

```diff
@@ -66,7 +66,9 @@ class PyConsoleCopyHandler(val originalHandler: EditorActionHandler) : EditorAct
       val rangeEnd = Math.min(document.getLineEndOffset(i), end)
       if (rangeStart < rangeEnd) {
         sb.append(document.getText(TextRange(rangeStart, rangeEnd)))
-        sb.append("\n")
+        if (rangeEnd < end) {
+          sb.append("\n")
```

Comment:
Yes, trim would accomplish the same thing. The only difference would be if the last selected character is a real '\n' .Not sure if it's actually possible.